torch/fx/passes/reinplace.py

```diff
@@ -0,0 +1,540 @@
+    assert inplace_schema.arguments[0].alias_info is not None and inplace_schema.arguments[0].alias_info.is_write
+    # and its remaining arguments shouldn't be.
+    assert all(a.alias_info is None for a in inplace_schema.arguments[1:])
+    return names_match and arg_types_match
+
+# TODO: this should be beefed up to be able to properly re-inplace with:
+# - mutating ops (e.g. _fused_moving_avg_obs_fq_helper)
+# - out= ops (e.g. angle -> angle.out)
+# TODO: we should also figure this info out using torchgen.
+def _maybe_get_inplace_op(op):
+    # __module__ seems broken; it returns torch._ops.aten which doesn't exist
+    if not isinstance(op, torch._ops.OpOverload):
+        return None
+    # Some view ops have inplace variants (as_strided_, etc),
+    # but we do NOT want the reinplacing pass to directly add these into the program.
+    # (they'll require extra special handling, aren't aren't really useful for perf anyway)
+    if _is_view_op(op):
+        return None
+    op_namespace = op.__module__.split(".")[-1]
+    op_base_name = op.overloadpacket.__name__
+    maybe_namespace_module = getattr(torch.ops, op_namespace)
+    maybe_inplace_op = None if maybe_namespace_module is None else getattr(maybe_namespace_module, f'{op_base_name}_', None)
+    if maybe_inplace_op is None:
+        return None
+
+    inplace_overloads = [
+        getattr(maybe_inplace_op, overload_name) for overload_name in maybe_inplace_op.overloads()
+    ]
+    inplace_overloads_with_matching_schemas = [
+        f
+        for f in inplace_overloads
+        if _schemas_match(op._schema, f._schema)
+    ]
+    # This is for sanity: if foo() and foo_() are both operators,
+    # we expect them to have compatible schemas.
+    # (This is asserted by codegen for ATen, but might not be true
+    # for other arbitrary operators).
+    assert len(inplace_overloads_with_matching_schemas) == 1
+    inplace_op = inplace_overloads_with_matching_schemas[0]
+    return inplace_op
+
+def _maybe_get_base_of_curr_node(self_arg):
+    # If the current arg was created from a view op, return the node
+    # corresponding to tensor that it was viewed from.
+    if _is_view_op(self_arg.target):
+        assert isinstance(node.args[0], torch.fx.node.Node)
+        return self_arg.args[0]
+    # We might need to look one level down:
+    # for multi-output view ops like .split(),
+    # The split op creates a list of tensors, and _operator.getitem
+    # is recorded into the FX graph to pass individual tensors from that list
+    # into other operators.
+    if node.target is _operator.getitem:
+        list_arg = self_arg.args[0]
+        if _is_view_op(list_arg.target):
+            assert isinstance(list_arg.args[0], torch.fx.node.Node)
+            return list_arg.args[0]
+    return None
+
+def _can_reuse_tensor_memory(self_arg, *, view_of_self: Optional[torch.fx.node.Node] = None):
+    nodes_used = self_arg.meta['node_usages'] if 'node_usages' in self_arg.meta else []
```

Comment:
is it possible for this field to not be set?

Reply:
oops, unused function that I forgot to delete